test/transport.test.js

```diff
@@ -0,0 +1,158 @@
+  t.ok(data[1].startsWith('<134>Feb 10 01:20:00 MacBook-Pro-3 none[94473]: '), true, 'first line trailingDay')
+})
+
+test('syslog pino transport test rfc5424', async t => {
+  const destination = join(os.tmpdir(), 'pino-transport-test.log')
+
+  const fd = fs.openSync(destination, 'w+')
+  const sysLogOptions = {
+    destination: fd,
+    ...getConfigPath('5424', 'newline.json')
+  }
+
+  const transport = pino.transport({
+    target: pinoSyslog,
+    level: 'info',
+    options: sysLogOptions
+  })
+  const log = pino(transport)
+  t.pass('built pino')
+  await once(transport, 'ready')
+  t.pass('transport ready ' + destination)
+
+  log.info(JSON.parse(messages.leadingDay))
+  log.debug(JSON.parse(messages.helloWorld)) // it is skipped
+  log.info(JSON.parse(messages.trailingDay))
+
+  await timeout(1000)
+
+  const data = fs.readFileSync(destination, 'utf8').trim().split('\n')
+  t.ok(data[0].startsWith('<134>1 2018-02-03T01:20:00Z MacBook-Pro-3 - 94473 - - '), 'first line leadingDay')
+  t.ok(data[1].startsWith('<134>1 2018-02-10T01:20:00Z MacBook-Pro-3 - 94473 - - '), 'first line trailingDay')
+})
+
+test('syslog pino transport test stdout', async t => {
+  const result = spawnSync('node', ['--no-warnings', join(__dirname, 'fixtures', 'log-stdout.js'), '1'], {
+    cwd: process.cwd()
+  })
+  t.equal(result.stdout.toString().trim(), '<134>1 2016-04-01T16:44:58Z MacBook-Pro-3 - 94473 - - hello world')
+  t.equal(result.status, 0)
+})
+
+test('syslog pino transport test stderr', async t => {
+  const result = spawnSync('node', ['--no-warnings', join(__dirname, 'fixtures', 'log-stdout.js'), '2'], {
+    cwd: process.cwd()
+  })
+  t.equal(result.stderr.toString().trim(), '<134>1 2016-04-01T16:44:58Z MacBook-Pro-3 - 94473 - - hello world')
+  t.equal(result.status, 0)
+})
+
+test('pino pipeline', t => {
+  t.plan(4)
+  const destination = join(os.tmpdir(), 'pino-transport-test.log')
+
+  const expected = [
+    '<134>1 2018-02-03T01:20:00Z MacBook-Pro-3 - 94473 - - ',
+    '<134>1 2018-02-10T01:20:00Z MacBook-Pro-3 - 94473 - - '
+  ]
+
+  createTcpListener(msg => {
+    msg.split('\n')
+      .filter(line => line) // skip empty lines
+      .forEach(line => {
+        t.ok(line.startsWith(expected.shift()))
+      })
+  })
+    .then((serverSocket) => {
+      t.teardown(() => {
+        serverSocket.close()
+        serverSocket.unref()
+        transport.end()
+      })
+
+      const address = serverSocket.address().address
+      const port = serverSocket.address().port
+
+      const transport = pino.transport({
+        pipeline: [
+          {
+            target: pinoSyslog,
+            level: 'info',
+            options: {
+              enablePipelining: true,
+              ...getConfigPath('5424', 'newline.json')
+            }
+          },
+          {
+            target: 'pino-socket',
+            options: {
+              mode: 'tcp',
+              address,
+              port
+            }
+          }
+        ]
```

Comment:
What would happen if a third transport is added? Are we saying that a transport with pipelining enabled will always pipe to the immediate next tansport in a transports array?

Reply:
Yes exactly. There is no fan-out, it's functionally equivalent to `cat a | grep something | cat -`.